pkg/chains/storage/gcs/gcs.go

```diff
@@ -111,7 +111,7 @@ func (b *Backend) StorePayload(ctx context.Context, obj objects.TektonObject, ra
 			return err
 		}
 	} else {
-		return fmt.Errorf("type %T not supported - supported types: [*v1beta1.TaskRun, *v1beta1.PipelineRun]", obj.GetObject())
+		return fmt.Errorf("type %T not supported - supported types: [*v1.TaskRun, *v1.PipelineRun]", obj.GetObject())
```

Comment:
For this and other storers - we want to preserve v1beta1 output, since this is going to be the visible change users see.

Reply:
This makes sense, I will update the comment thread here when I make this change. OOC, when a v2alpha3 is introduced that attempts to natively use v1 for the objects do we have an understanding upon how updating or adding additional logic here might impact users?  In general the Storage interface is mapped to the version and I'm a bit confused how tightly coupled it is from the UX perspective.  Additionally when we do a read from storage is there any metadata stating the GVK of the stored Tekton Object for storage types that seem to supply this info?  Some misc notes I took below:

- Storage interface takes a TektonObject obj which is versioned -> `StorePayload(ctx context.Context, obj objects.TektonObject, rawPayload []byte, signature string, opts config.StorageOpts) error`
- GVK stored as in grafeas storage opts
- GCS seems to use versioned (eg: v1.[TaskRun|Pipeline] references) Tekton Pipeline libs
  - `RetrievePayload` might need to be more intelligent about versions if we want to support backwards compat - https://github.com/tektoncd/chains/blob/main/pkg/chains/storage/gcs/gcs.go#L177

Reply:
In manually testing things, I believe the `gcs.go` (and `gcs_test.go`) code has to be updated to support v1 objects.  StorePayload only receives v1 objects with this change so it doesn't make sense to have it support v1beta1 objects.  The log message there will change but it isn't clear to me this is an issue (we can keep the text the same if desired but functionally StorePayload needs to use v1 objects.  This seems better than converting v1 objects to v1beta1 and using them which is more verbose and complicated and isn't necessary as noly Name, Namespace and UID are used from the objects).  The actual objects aren't used, only the Name, Namespace, and UID.  I have added back modifications to gcs.go and gcs_test.go

Reply:
Just want to note, the code will work for v1beta1 objects, it's just that they are supported via the Tekton ConversionWebhook.  This will still work for users as it did, it's just this one log line happens and the change is correct due to the internals change going v1beta1 -> v1

Reply:
Makes sense! Let's add a comment explaining this to the function godoc.